worker/uniter/operation/state.go

```diff
@@ -195,35 +200,51 @@ func (change stateChange) apply(state State) *State {
 	return &state
 }
 
-// StateFile holds the disk state for a uniter.
-type StateFile struct {
-	path string
+// StateOps reads and writes uniter state to/from the controller.
+type StateOps struct {
+	UniterStateOps
```

Comment:
How about renaming `UniterStateOps` to `UniterStateReadWriter`?
Embedding `UniterStateOps` into `StateOps` took me a while to parse...

Also, why embed instead of assigning to a variable? Are we calling any of the `UniterStateOps` methods on a `StateOps` instance?

Reply:
Renaming and using a variable, the interface should not be used directly.